console/src/main/scala/io/shiftleft/console/query/DefaultCpgQueryExecutor.scala

```diff
@@ -2,27 +2,32 @@ package io.shiftleft.console.query
 
 import cats.data.OptionT
 import cats.effect.{Blocker, ContextShift, IO}
-import javax.script.ScriptEngineManager
-
+import javax.script.{ScriptEngine, ScriptEngineManager}
 import io.shiftleft.codepropertygraph.Cpg
-
 import java.util.UUID
 import java.util.concurrent.{ConcurrentHashMap, Executors}
 
 import scala.collection.JavaConverters._
 import scala.collection.concurrent.Map
 import scala.concurrent.ExecutionContext
 
+/**
+  * This class executes a query on a given CPG.
+  */
 class DefaultCpgQueryExecutor(scriptEngineManager: ScriptEngineManager)(implicit val cs: ContextShift[IO])
-    extends CpgQueryExecutor[String] {
+    extends CpgQueryExecutor[AnyRef] {
+
+  private val ENGINE_NAME = "scala"
 
-  private val engineType = "scala"
+  private lazy val engine: IO[ScriptEngine] = OptionT
+    .fromOption[IO](Option(scriptEngineManager.getEngineByName(ENGINE_NAME)))
+    .getOrElseF(IO.raiseError(new RuntimeException("Engine could not be instantiated.")))
```

Comment:
This can be a plain `val`. `IO` instances are only evaluated once they are `run`. We can probably inline the `ENGINE_NAME` on L20, too as it's only used here.

Reply:
Fixed.